### PR TITLE
prevent exception when destroying vim plugin

### LINKED ISSUE #46: Can't change EditorState using "Vim.defineEx"
Description
---
As the titled written, I can't change EditorState using [`Vim.defineEx`](https://codemirror.net/5/doc/manual.html#vimapi_extending). Usually I want to preserve extension between different EditorState, so I am using [`StateEffect`](https://codemirror.net/docs/ref/#state.StateEffect). I thought that was the problem, so I tried to disabled that and get the same error message. While when using `<button>` to change EditorState, it's work without any error.  

With Button  
---
With Button

With Vim Command
---
With Command

Code Sandbox Link
---
[Code Sandbox](https://codesandbox.io/s/cant-change-editor-state-using-vim-command-69od3u)
[Code Sandbox (without preserving extension)](https://codesandbox.io/s/cant-change-editor-state-using-vim-command-simple-ob9h2h)

### PR DESCRIPTION
**Pull request by nightwing**
fixes #46, #47

### CODE REVIEW

**Review by lunaroyster — COMMENTED** [review]
looks good to me, and fixes some major issues, thank you! 

i think @sergeichestakov or @masad-frost should provide the approving review since they understand the architecture better

**Review by masad-frost — APPROVED** [review]
:shipit:

File: dev/index.ts [hunk 1]
```diff
@@ -59,25 +60,76 @@ global._commands = commands;
...
+  // make sure vim is included before all other keymaps
+  vimCompartement.of([
+    vim({status: true}),
+  ]),
+  // include the default keymap and all other keymaps you want to use in insert mode
+  basicSetup,
+  highlightActiveLine(),
+  keymap.of([
+    {
+      key: "Alt-v",
```
Comment (sergeichestakov) [inline dev/index.ts:76]: why does `Alt-v` disable / enable Vim?

File: dev/index.ts [hunk 2]
```diff
@@ -59,25 +60,76 @@ global._commands = commands;
 global._Vim = Vim;
 
 let view
+let enableVim = true
```
Comment (sergeichestakov) [inline dev/index.ts:63]: can we dispatch an effect to toggle the extension from the compartment rather than relying on globals?

**Review by sergeichestakov — CHANGES_REQUESTED** [review]
Thank you! This sounds reasonable but can we toggle the extension via an effect rather than using globals? See this example in the [CM docs](https://codemirror.net/examples/config/#private-compartments): 

Screen Shot 2022-10-04 at 5 07 28 PM

**Comment by masad-frost** [comment]
@sergeichestakov this is just for the dev/example, i think using globals is fine there.

**Comment by sergeichestakov** [comment]
Fair enough. Your call then! Might be nice to keep the example clean and according to best practices tho for others to get a better idea of how to use imo but no strong feelings

**Review by sergeichestakov — APPROVED** [review]

**Pull request merged**